fda-model/stageDamage/ImpactAreaStageDamage.cs

```diff
@@ -313,21 +313,22 @@ public List<UncertainPairedData> Compute(IProvideRandomNumbers randomProvider)
             }
             else
             {
-                List<double> allStagesAtIndexLocation = new List<double>();
-                List<ConsequenceDistributionResults> consequenceDistributionResults = new List<ConsequenceDistributionResults>();
                 PairedData stageFrequency = CreateStageFrequency();
-
-
-            List<string> damCats = _inventory.DamageCategories;
-            foreach (string damageCategory in damCats)
-            {
-                Inventory inventory = _inventory.GetInventoryTrimmedToDamageCategory(damageCategory);
-                ComputeLowerStageDamage(inventory, randomProvider, stageFrequency, damageCategory, ref allStagesAtIndexLocation, ref consequenceDistributionResults);
-                ComputeMiddleStageDamage(inventory, randomProvider, stageFrequency, damageCategory, ref allStagesAtIndexLocation, ref consequenceDistributionResults);
-                ComputeUpperStageDamage(inventory, randomProvider, stageFrequency, damageCategory, ref allStagesAtIndexLocation, ref consequenceDistributionResults);
-            }
-
-                results = ConsequenceDistributionResults.ToUncertainPairedData(allStagesAtIndexLocation, consequenceDistributionResults, _ImpactAreaID);
+                List<string> damCats = _inventory.DamageCategories;
+                foreach (string damageCategory in damCats)
+                {
+                    List<double> allStagesAtIndexLocation = new List<double>();
+                    List<ConsequenceDistributionResults> consequenceDistributionResults = new List<ConsequenceDistributionResults>();
+                    Inventory inventory = _inventory.GetInventoryTrimmedToDamageCategory(damageCategory);
+                    ComputeLowerStageDamage(inventory, randomProvider, stageFrequency, damageCategory, ref allStagesAtIndexLocation, ref consequenceDistributionResults);
+                    ComputeMiddleStageDamage(inventory, randomProvider, stageFrequency, damageCategory, ref allStagesAtIndexLocation, ref consequenceDistributionResults);
+                    ComputeUpperStageDamage(inventory, randomProvider, stageFrequency, damageCategory, ref allStagesAtIndexLocation, ref consequenceDistributionResults);
+                    List<UncertainPairedData> tempResultsList = ConsequenceDistributionResults.ToUncertainPairedData(allStagesAtIndexLocation, consequenceDistributionResults, _ImpactAreaID);
+                    foreach (UncertainPairedData uncertainPairedData in tempResultsList)
```

Comment:
I think this can be simplified by doing results.AddRange(tempResultsList);

Reply:
done